dlc/src/test/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelfIntegrationTest.scala

```diff
@@ -220,28 +230,60 @@ class BinaryOutcomeDLCWithSelfIntegrationTest extends BitcoindRpcTest {
 
     def executeForRefundCase(): Future[Assertion] = {
       for {
-        client <- clientF
         dlc <- constructDLC()
         setup <- dlc.setupDLC()
         _ <- publishTransaction(setup.fundingTx)
         outcome <- dlc.executeRefundDLC(setup)
-        currentCount <- client.getBlockCount
-        blocksToMine = dlc.timeout.toUInt32.toInt - currentCount
-        addressForMining <- addressForMiningF
-        _ <- client.generateToAddress(blocks = blocksToMine, addressForMining)
+        _ <- waitUntilBlock(dlc.timeout.toUInt32.toInt)
         _ <- publishTransaction(outcome.cet)
         _ <- publishTransaction(outcome.localClosingTx)
         _ <- publishTransaction(outcome.remoteClosingTx)
         assertion <- validateOutcome(outcome)
       } yield assertion
     }
 
+    def executeForJusticeCase(
+        fakeWin: Boolean,
+        local: Boolean): Future[Assertion] = {
+      def chooseCET(setup: SetupDLC): Transaction = {
+        if (fakeWin) {
+          if (local) {
+            setup.cetWinRemote
+          } else {
+            setup.cetWinLocal
+          }
+        } else {
+          if (local) {
+            setup.cetLoseRemote
+          } else {
+            setup.cetLoseLocal
+          }
+        }
+      }
+
+      for {
+        dlc <- constructDLC()
+        setup <- dlc.setupDLC()
+        _ <- publishTransaction(setup.fundingTx)
+        outcome <- dlc.executeJusticeDLC(setup, chooseCET(setup), local)
```

Comment:
so i think i am misunderstanding something here, the justice transaction should be spending from a CET right? From my understanding of this, we publish the funding transaction, and then attempt to spend from the funding transaction with a justice transaction? 

If I look below this, it does look like a CET is published, maybe what might make this clearer to me is if we publish a CET before determining the dlc outcome? Isn't that what would actually happen in the real world? If we are publishing a justice transaction, that presumably means that the counter party has cheated no?

Reply:
I'll reorder, but just to clarify things for you, a justice tx is not a CET, it spends a CET. Just like in lightning a justice tx isn't a commitment tx it spends one